### DISCUSSION TITLE
Job Applicant Tracker - Session Expired or Don't Have Permissions

### DISCUSSION THREAD

**Discussion by andrewijm** [Budibase/budibase#3789] [Help] [answered]
Really struggling with the fundamentals here... hoping for some help.

1. Created new Job Applicant Tracker App. 
2. Published App.
3. Opened in Incognito Mode.

This is the error I get. Why would public user not have access, or why would the session be expired?
![Screen Shot 2021-12-15 at 11 28 36 AM](https://user-images.githubusercontent.com/40970684/146103363-375800d8-5fb6-4924-ac29-4f7f73971b06.jpg)

**Comment by almereyda** [comment, marked as answer]
I'm seeing the same with a membership form.

I could circumvent the issue by not just allowing write, but also by allowing read access to the desired table. Then the client could fetch the schema information about the table from the database, and go on as expected, without a warning.

**Reply by andrewijm** [reply]
That did the trick! Thanks.

**Reply by almereyda** [reply]
Yes, after I wrote you I've found that this is also mentioned and explained in step three of https://docs.budibase.com/guides/creating-public-and-private-screens#guide :

> - [ ] Change read and write access to Public

plus further (emphasis mine)

> And now we have a public screen. In our case, public users can now access our leads/new/row screen which has our new leads form. They can also **complete the form and submit it**. The form requires some styling but hopefully, you now understand the concept and how to create screens based on the user's role (public, basic, power, admin).
>
> Once a public user submits their information, **we and other private (basic, power, admin) users can view the leads and their details within the private section of our app**.

I'm not sure if it's a good design decision to have access control on the view level, rather than at the table (API) level. Maybe Budibase can improve here and offer to make a table's metadata public, for such discovery requests (Why is it even there?), next to public write access for submitting such forms, yet remain private with reading the entries themselves?

I have run into this issue (with setting tables to write for public, but only to read for private) multiple times, and would want to know a standard way to avoid it, which potentially does not expose private data to the outside.

**Reply by andrewleith** [reply]
I had the same question.  I wanted to try out the job application app and encourage some colleagues to use it to manage their job postings.  First thing I noticed was this error for an unauthenticated user.

The problem here is twofold:

1. The app templates should work properly out-of-the-box, with no errors.
2. It seems like the fundamental setup here is a security flaw - why would `public` users need to be able to read data from this table at all? What in budibase is triggering this error?  There is no explicit reading from the table in this screen. This seems like an easy way for someone to inadvertently leak private data.

**Reply by almereyda** [reply]
Exactly, by design.

**Comment by almereyda** [comment]
It appears, which lacks testing, that public forms are now available:

- https://docs.budibase.com/docs/public-forms
